ocaml/quicktest/qt_filter.ml

```diff
@@ -128,7 +128,24 @@ module SR = struct
 
   let only sr () = [get_sr_info !A.rpc !session_id sr]
 
-  let all_srs = lazy (list_selected_srs !A.rpc !session_id)
+  let all_srs =
+    let is_shared_or_local sr_info =
+      (** allows us to ignore srs which don't belong to us.
+        * this remedies some failing tests if we run the quicktest binary on a host in a pool *)
+      let sr = sr_info.Qt.sr in
+      let session_id = !session_id in
+      let rpc = !A.rpc in
+      let pbds = Client.Client.SR.get_PBDs ~rpc ~session_id ~self:sr in
+      match pbds with
+      | [pbd] -> begin
```

Comment:
The `begin`/`end` is not required for the right-hand side of a `match` case.

Reply:
I think it's because of how it interacts with `list_selected_srs`, but I think @lippirk it would be cleaner to merge this functionality into `list_selected_srs`, because it seems that function is only used from here, and maybe rename it to something more descriptive, like `list_available_srs`.